src/common/convauto.cpp

```diff
@@ -307,22 +307,19 @@ wxConvAuto::ToWChar(wchar_t *dst, size_t dstLen,
 
     // try to convert using the auto-detected encoding
     size_t rc = m_conv->ToWChar(dst, dstLen, src, srcLen);
-    if ( rc == wxCONV_FAILED && m_bomType == wxBOM_None )
+    if ( rc == wxCONV_FAILED && m_bomType == wxBOM_None && !m_ownsConv )
     {
         // we may need more bytes before we can decode the input, don't switch
         // to the fall-back conversion in this case as it would prevent us from
         // decoding UTF-8 input when fed it byte by byte, as done by
         // wxTextInputStream, for example
-        if ( srcLen < m_conv->GetMaxCharLen() )
+        if ( srcLen < 2 + m_conv->GetMaxCharLen() && wxIsUTF8Prefix(src, srcLen) )
```

Comment:
Could you please add a comment here explaining this `+ 2`? I don't really understand where does it come from and what would break if we didn't have it.